packages/services/api/src/modules/schema/providers/schema-publisher.ts

```diff
@@ -384,8 +386,8 @@ export class SchemaPublisher {
         schemaPolicyWarnings: checkResult.state?.schemaPolicyWarnings ?? null,
         schemaPolicyErrors: null,
         schemaCompositionErrors: null,
-        compositeSchemaSDL: composition.compositeSchemaSDL,
-        supergraphSDL: composition.supergraphSDL,
+        compositeSchemaSDL: hasSchemaChanges ? composition.compositeSchemaSDL : null,
+        supergraphSDL: hasSchemaChanges ? composition.supergraphSDL : null,
```

Comment:
in case there has been no initial schema published the SDL and supergraph need to be stored as well. (schemaVersionId` is `null`).

Reply:
can do, but does it makes sense?

Reply:
Yes, it makes sense because otherwise there is no compositeSchemaSDL/supergraphSDL that can be displayed. It is an edge-case that should be handled correctly.

Reply:
I tested this case - empty federation project, it gives me this and no schema checks are created. 

<img width="492" alt="Screenshot 2023-08-21 at 17 54 38" src="https://github.com/kamilkisiela/graphql-hive/assets/78734704/c6744510-afe1-4f0f-8dcb-2f3f32c4022f">